packages/beacon-node/src/chain/opPools/aggregatedAttestationPool.ts

```diff
@@ -319,15 +346,19 @@ export class AggregatedAttestationPool {
       const epoch = computeEpochAtSlot(slot);
       // validateAttestation condition: Attestation target epoch not in previous or current epoch
       if (!(epoch === stateEpoch || epoch === statePrevEpoch)) {
-        continue; // Invalid attestations
+        // we process slot in desc order, this means slot is out of current or previous epoch, we should stop
```

Comment:
so realistically this happens only if slot is out of previous epoch, or is there an edge case were this could also be current epoch?

Reply:
> so realistically this happens only if slot is out of previous epoch, or is there an edge case were this could also be current epoch?

The code here assumes `slots` starts with `stateEpoch` and then goes in descending order.

I was thinking if an edge case like `slots` start with `stateEpoch + 1` could happen. If that happens, the loop will break in the first iteration but I don't think that's possible.

Reply:
I see the concern, will separate to a separate if condition
```typescript
if (epoch < statePrevEpoch) {
  // we process slot in desc order, this means next slot is not eligible, we should stop
  stopReason = ScannedSlotsTerminationReason.SlotBeforePreviousEpoch;
  break;
}
```

Reply:
this makes sense I will wonder how this condition can be true though, unless our pruning didn't work as expect, ie. due to clock being off?

Reply:
it's not supposed to reach that condition but the code is not redundant, we need to handle different flows even it may not happen. In case things go wrong we know where to look at through that metric

Reply:
yeah I agree, it makes sense as sanity check / assert here